pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -452,8 +452,9 @@ func PrepareLogsQuery(start, end int64, queryType v3.QueryType, panelType v3.Pan
 		} else {
 			query = addLimitToQuery(query, mq.Limit)
 		}
+	} else if panelType == v3.PanelTypeTable {
+		query = addLimitToQuery(query, mq.Limit)
```

Comment:
How does this fix the issue? There is an else clause after `if mq.PageSize > 0`, which is doing the same thing. Is pageSize non-zero for table type?